packages/react-core/src/components/DataList/__tests__/DataListToggle.test.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { DataListToggle } from '../DataListToggle';
+
+import styles from '@patternfly/react-styles/css/components/DataList/data-list';
+
+test('DataListAction renders to match snapshot', () => {
+  const { asFragment } = render(<DataListToggle id="ex-toggle2" />);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders with default class ${styles.dataListToggle}`, () => {
+  render(<DataListToggle id="ex-toggle2" />);
+  expect(screen.getByRole('button').parentElement).toHaveClass(styles.dataListToggle);
+});
+
+test(`Renders with default class ${styles.dataListItemControl}`, () => {
+  render(<DataListToggle id="ex-toggle2" />);
+  expect(screen.getByRole('button').parentElement?.parentElement).toHaveClass(styles.dataListItemControl);
+});
+
+test(`Renders with custom class when className is passed`, () => {
+  render(<DataListToggle className="custom" id="ex-toggle2" />);
+  expect(screen.getByRole('button').parentElement?.parentElement).toHaveClass('custom');
+});
+
+test('Renders with aria label', () => {
+  render(<DataListToggle aria-label="Toggle details for" id="ex-toggle2" />);
+
+  expect(screen.getByRole('button')).not.toHaveAttribute('aria-labelledby');
```

Comment:
We could update this test name to "Does not render aria-labelledby when custom aria-label is passed", then add another test to check that aria-labelledby renders by default if aria-label is not passed

Shoudl also have a test for aria-controls and that buttonProps spreads to Button correctly (could mock the button or just check the button role has some attribute; I did the latter in the clipboard copy test update PR fwiw)